File: App/Resources/AlgorithmDetails/lessbogosort.bundle/lessbogosort.java
import java.util.Arrays;

public class lessbogosort {
  public static boolean isMinimum(int[] arr, int start, int end) {
    for (int k = start + 1; k < end; k++) {
      if (arr[start] > arr[k]) {
        return false;
      }
    }
    return true;
  }

  public static void shuffleRange(int[] arr, int start, int end) {
    for (int i = start; i < end - 1; i++) {
      int j = i + (int)(Math.random() * (end - i));
      int t = arr[i];
      arr[i] = arr[j];
      arr[j] = t;
    }
  }

  public static void sort(int[] arr) {
    int n = arr.length;
    for (int i = 0; i < n; i++) {
      while (!isMinimum(arr, i, n)) {
        shuffleRange(arr, i, n);
      }
    }
  }

  public static void main(String[] args) {
    int[] array = new int[]{0, 39, 21, 62, 91, 77, 14, 23};
    sort(array);
    System.out.println(Arrays.toString(array));
  }
}

File: App/Resources/AlgorithmDetails/lessbogosort.bundle/lessbogosort.kt
fun isMinimum(arr: Array<Int>, start: Int, end: Int): Boolean {
  for (k in start + 1 until end)
    if (arr[start] > arr[k])
      return false
  return true
}

fun shuffleRange(arr: Array<Int>, start: Int, end: Int) {
  for (i in start until end - 1) {
    val j = (i until end).random()
    val t = arr[i]
    arr[i] = arr[j]
    arr[j] = t
  }
}

fun sort(arr: Array<Int>) {
  val n = arr.size
  for (i in 0 until n)
    while (!isMinimum(arr, i, n))
      shuffleRange(arr, i, n)
}

fun main() {
  val array = arrayOf<Int>(0, 39, 21, 62, 91, 77, 14, 23)
  sort(array)
  println("[%s]".format(array.joinToString(", ")))
}
